pkg/ioutils/readers.go

```diff
@@ -50,11 +52,11 @@ func NewReaderErrWrapper(r io.Reader, closer func()) io.Reader {
 	}
 }
 
-// bufReader allows the underlying reader to continue to produce
+// BufReader allows the underlying reader to continue to produce
 // output by pre-emptively reading from the wrapped reader.
-// This is achieved by buffering this data in bufReader's
+// This is achieved by buffering this data in BufReader's
 // expanding buffer.
-type bufReader struct {
+type BufReader struct {
```

Comment:
I think this can just return a `io.ReadCloser` since all fields on BufReader are private and the only exported functions are `Read` and `Close`.